dnf-behave-tests/createrepo_c/environment.py

```diff
@@ -85,7 +91,9 @@ def after_tag(context, tag):
 
 
 def before_all(context):
-    pass
+    os = context.config.userdata.get("os", None)
+    if os:
+        active_tag_value_provider["os"] = os
```

Comment:
Couldn't you just do something like:
```python
context.tag_matcher = VersionedActiveTagMatcher({"os": context.config.userdata.get("os", UNDEFINED)})
```
Where `UNDEFINED` is some default value representing an unidentified os. If the tag matcher won't work with this (I haven't checked...), just replace this oneliner with constructing the dict and adding the "os" item only if it's defined...

Then you can just use `context.tag_matcher` in `before_scenario()`.

This is cleaner and saves us the global variables, which don't hurt in this case but it's good practice to avoid global variables if you can.

Reply:
I have updated it this way also in dnf `environment.py`.